app/src/main/java/in/co/rajkumaar/amritarepo/aums/activities/CourseResourcesActivity.java

```diff
@@ -0,0 +1,496 @@
+                            String resType;
+                            String oneUp = "../";
+                            if (!(resUrl.equals(oneUp))) {
+                                if (resUrl.substring(resUrl.length() - 1).equals("/")) {
+                                    resType = "Folder";
+                                } else {
+                                    resType = resUrl.substring(resUrl.lastIndexOf('.') + 1);
+                                }
+                                resUrl = folder + resUrl;
+                                courseResourceList.add(new CourseResource(resName, resUrl, resType));
+                                if (resName.equals(lastRes)) {
+                                    courseResourceStack.add(courseResourceList);
+                                    CourseResAdapter courseResAdapter = new CourseResAdapter(CourseResourcesActivity.this, courseResourceList);
+                                    list.setAdapter(courseResAdapter);
+                                    if (firstEntry) {
+                                        findViewById(R.id.progressBar).setVisibility(GONE);
+                                        list.setVisibility(View.VISIBLE);
+                                        firstEntry = false;
+                                    } else {
+                                        progressDialog.dismiss();
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } catch (Exception e) {
+                    Toast.makeText(CourseResourcesActivity.this, getString(R.string.site_change), Toast.LENGTH_LONG).show();
+                    finish();
+                }
+            }
+
+            @Override
+            public void onFailure(int statusCode, Header[] headers, byte[] responseBody, Throwable error) {
+                Utils.showUnexpectedError(CourseResourcesActivity.this);
+                finish();
+            }
+        });
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (courseResourceStack.size() == 1) {
+            finish();
+        } else {
+            int index = courseResourceStack.size() - 1;
+            CourseResAdapter courseResAdapter = new CourseResAdapter(CourseResourcesActivity.this, courseResourceStack.get(index - 1));
+            list.setAdapter(courseResAdapter);
+            courseResourceStack.remove(index);
+
+            setTitle(curFolder.get(index - 1));
+            curFolder.remove(index);
+        }
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                if (courseResourceStack.size() > 1) {
+                    onBackPressed();
+                } else {
+                    NavUtils.navigateUpFromSameTask(this);
```

Comment:
```suggestion
                    finish();
```
so that the previous activity from android stack doesn't reload again.